arch/cortex-m/src/dcb.rs

```diff
@@ -0,0 +1,58 @@
+//! ARM Debug Control Block
+//!
+//! <https://developer.arm.com/documentation/ddi0403/latest>
+
+use kernel::utilities::registers::interfaces::ReadWriteable;
+use kernel::utilities::registers::{register_bitfields, register_structs, ReadWrite};
+use kernel::utilities::StaticRef;
+
+//TODO: Rest of the Registers
```

Comment:
<img width="849" alt="image" src="https://user-images.githubusercontent.com/339422/192852127-435212c5-55fd-43aa-884d-e9451375da88.png">

There's only three more here; I think we'd ask for the register definition to be fully implemented before pulling this in (i.e., flesh out the rest of `register_structs` and `register bitfields`; don't need all the getters/setters/syscalls at first).